benchmarks/benchmarks/stats.py

```diff
@@ -57,45 +57,107 @@ def time_ttest_ind_diff_var(self):
 
 
 class Distribution(Benchmark):
+    # all distributions are in this list. A conversion to a set is used to
+    # remove duplicates that are files under `distcont` and `distdiscrete`.
+    dists = sorted(list(set([d[0] for d in distcont + distdiscrete])))
+
     param_names = ['distribution', 'properties']
     params = [
-        ['cauchy', 'gamma', 'beta'],
-        ['pdf', 'cdf', 'rvs', 'fit']
+        dists, ['pdf/pmf', 'logpdf/logpmf', 'cdf', 'logcdf', 'rvs', 'fit',
+                'sf', 'logsf', 'ppf', 'isf', 'moment', 'stats_s', 'stats_v',
+                'stats_m', 'stats_k', 'stats_mvsk', 'entropy']
     ]
+    distcont = dict(distcont + distdiscrete)
+    # maintain previous benchmarks' values
+    custom_input = {'gamma': [5], 'beta': [5, 3]}
+
+    # these distributions are the slowest quartile and are only run if is_xslow
+    # is set
+    slow_dists = ['alpha', 'argus', 'cosine', 'exponnorm', 'exponweib',
+                  'f', 'foldcauchy', 'foldnorm', 'genlogistic', 'genexpon',
+                  'gausshyper', 'gumbel_r', 'halfgennorm', 'invgauss',
+                  'johnsonsu', 'kappa4', 'ksone', 'kstwo', 'levy_stable',
+                  'lognorm', 'ncx2', 'nct', 'pearson3', 'powerlognorm',
+                  'powernorm', 'recipinvgauss', 'vonmises',
+                  'vonmises_line', 'wald']
+    slow_methods = ['moment']
 
     def setup(self, distribution, properties):
-        np.random.seed(12345678)
-        self.x = np.random.rand(100)
+
+        if not is_xslow() and (distribution in self.slow_dists or
+                               properties in self.slow_methods):
+            raise NotImplementedError("Skipped")
+
+        self.dist = getattr(stats, distribution)
+
+        shapes = self.distcont[distribution]
+
+        if isinstance(self.dist, stats.rv_discrete):
+            self.isCont = False
+            kwds = {'loc': 4}
+        else:
+            self.isCont = True
+            kwds = {'loc': 4, 'scale': 10}
+
+        rng = self.dist.interval(.99, *shapes, **kwds)
```

Comment:
```suggestion
        bounds = self.dist.interval(.99, *shapes, **kwds)
```
To avoid confusion with "random number generator"

Reply:
I guess it's not clear enough that it's supposed to mean 'range'. Bounds Is better!